src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,990 @@
+
+
+def retrieve_arc_agents_logs(corev1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    global diagnoser_output
+    try:
+
+        if storage_space_available:
+            # To retrieve all of the arc agents pods that are present in the Cluster
+            arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+            # Traversing through all agents
+            for each_agent_pod in arc_agents_pod_list.items:
+
+                # Fethcing the current Pod name and creating a folder with that name inside the timestamp folder
+                agent_name = each_agent_pod.metadata.name
+                arc_agent_logs_path = os.path.join(filepath_with_timestamp, "Arc_Agents_logs")
+                try:
+                    os.mkdir(arc_agent_logs_path)
+                except FileExistsError:
+                    pass
+
+                agent_name_logs_path = os.path.join(arc_agent_logs_path, agent_name)
+                try:
+                    os.mkdir(agent_name_logs_path)
+                except FileExistsError:
+                    pass
+
+                # If the agent is not in Running state we wont be able to get logs of the containers
+                if(each_agent_pod.status.phase != "Running"):
+                    continue
+
+                # Traversing through all of the containers present inside each pods
+                for each_container in each_agent_pod.spec.containers:
+
+                    # Fetching the Container name
+                    container_name = each_container.name
+
+                    # Creating a text file with the name of the container and adding that containers logs in it
+                    container_log = corev1_api_instance.read_namespaced_pod_log(name=agent_name, container=container_name, namespace="azure-arc")
+
+                    # Path to add the arc agents container logs.
+                    arc_agent_container_logs_path = os.path.join(agent_name_logs_path, container_name + ".txt")
+                    with open(arc_agent_container_logs_path, 'w+') as container_file:
+                        container_file.write(str(container_log))
+
+        return consts.Diagnostic_Check_Passed, storage_space_available
+
+    # For handling storage or OS exception that may occur during the execution
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An exception has occured while trying to fetch the azure arc agents logs from the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Arc_Agents_Logger_Fault_Type, summary="Error occured in arc agents logger")
```

Comment:
Actually, we can prefix all fault_types related to diagnoser with "diagnoser_". This will help in seggregating them and using for analysis purposes easily. Please add a comment saying "Diagnoser related fault types" and keep them at one place under this in consts.py file